src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.Timer;
+
+import com.ctre.phoenix6.controls.VelocityVoltage;
+import com.ctre.phoenix6.hardware.TalonFX;
+
+public class Intake extends SubsystemBase {
+    private final DigitalInput preIntakeSensor = new DigitalInput(0);
+    private final DigitalInput intakeSensor = new DigitalInput(1);
+    private final DigitalInput shooterSensor = new DigitalInput(2);
+
+    private final TalonFX intakeMotor = new TalonFX(0);
+
+    private boolean status = intakeMotor.isAlive();
+    // TODO: Change intakeMotorVelocity
+    private double intakeMotorVelocity = 1;
+    private VelocityVoltage velocity = new VelocityVoltage(intakeMotorVelocity);
+
+    private enum IntakeState {
+        IDLE, // No motors are moving, no note is inside the mechanism
+        INTAKING, // Motors are moving, note is not yet where we need it to be
+        HOLDING, // No motors are moving, note is where it needs to be and is contained in the robot
+        EJECTING; // Motors are moving, note is being moved into the gears
+    }
+    private Timer timer = new Timer();
+    
+
+    private static IntakeState currentState = IntakeState.IDLE;
+
+    @Override
+    public void periodic() {
+        SmartDashboard.putBoolean("Motor Functional?", status);
+        SmartDashboard.putNumber("Intake Motor Speed (RPM)", intakeMotorVelocity);
```

Comment:
Just FYI this is called every 20 ms and is probably fine but you may get some loop overrun stuff. It's fine to test and figure that out later.